File: RedSheet.DbEntities/Models/vPowerTypeDetailRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vPowerTypeDetailRecords", Schema = "dbo")]
	[Serializable]
	public partial class vPowerTypeDetailRecord  
	{
 
	[Range(1, int.MaxValue)]
	public int ActualId { get; set; } 
 
	[Required]
	public bool IsOurKnowledge { get; set; } 
 
	[Required]
	public bool IsTheirKnowledge { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int PowerTypeDetailId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int PowerTypeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string PowerTypeName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectedId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectPowerId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Rationale { get; set; } 

	public vPowerTypeDetailRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-power-type-detail-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vPowerTypeDetailRecord {
    constructor(vPowerTypeDetailRecord?: vPowerTypeDetailRecord )  {
        let properties = [ "actualId", "isOurKnowledge", "isTheirKnowledge", "powerTypeDetailId", "powerTypeId", "powerTypeName", "projectedId", "projectPowerId", "rationale",];
        for (let property of properties)
            if (vPowerTypeDetailRecord && vPowerTypeDetailRecord[property])
                this[property] = vPowerTypeDetailRecord[property];
    }
 
    @range(1,2147483647)
	actualId : number =   undefined;
 
	isOurKnowledge : boolean = false ;
 
	isTheirKnowledge : boolean = false ;
 
	powerTypeDetailId : number =   0 ;
 
    @range(1,2147483647)
	powerTypeId : number =   undefined;
 
    @required()
    @maxLength(200)
	powerTypeName : string =   undefined;
 
    @range(1,2147483647)
	projectedId : number =   undefined;
 
    @range(1,2147483647)
	projectPowerId : number =   undefined;
 
    @required()
    @maxLength(1000)
	rationale : string =   undefined;


}
